game/src/main/scala/io/aigar/game/Cell.scala

```diff
@@ -33,13 +39,22 @@ class Cell(id: Int, startPosition: Vector2 = new Vector2(0f, 0f)) {
   def velocity_=(vel:Vector2) {
     _velocity = if (vel.magnitude < maxSpeed) vel else vel.normalize * maxSpeed
   }
+  def mass = _mass
+  def mass_=(m: Float) {
+    _mass = max(m, Cell.MinMass)
+  }
 
 
   def update(deltaSeconds: Float) {
+    mass -= massDecay(deltaSeconds)
     velocity += acceleration * deltaSeconds
     position += velocity * deltaSeconds
   }
 
+  def massDecay(deltaSeconds: Float) = {
+    mass * pow(Cell.MassDecayPerSecond, deltaSeconds).toFloat
```

Comment:
Oops, you are totally right */facepalm*! It would work if we did `mass *= 0.995^time`, but `mass -= mass * (1 - 0.005) ^ time` is not the same as `mass -= mass * (1 - 0.005^time)` (what this PR does), so that's why it doesn't make sense.

I don't think we just want `decay_per_s * delta_s`. We're really in a case of a typical [growth and decay](http://www.regentsprep.org/regents/math/algebra/ae7/expdecayl.htm) problem (similar to bacteria dying over time, for example).

Basically, the reason why we included *time* in our calculations is so that the amount of times that we call `update` **doesn't matter**, all that matters is that the time sum is the same. In code, we want:
```
cell.update(1f)
```
to give the same result as:
```
cell.update(0.5f)
cell.update(0.5f)
```

Now, that doesn't work for our physics simulation (we can talk about it offline, we're doing an integration approximation here), but we can definitely make it work with mass decay.

If we were to only do `mass -= mass * mass_decay * delta_s`, here's what would happen with `delta_s = 1`:
```
mass = 1000 - (1000 * 0.005 * 1)
     = 1000 - 5
     = 995
```
With `delta_s = 0.5` applied twice:
```
mass' = 1000 - (1000 * 0.005 * 0.5)
      = 1000 - 2.5
      = 997.5
mass = 997.5 - (997.5 * 0.005 * 0.5)
     = 997.5 - 2.49375
     = 995.00625
```

While with `mass *= (1 - decay) ^ time`, for `delta_s = 1`:
```
mass = 1000 * (1 - 0.005)^1
     = 995
```
For `delta_s = 0.5` twice:
```
mass' = 1000 * (1 - 0.005)^0.5
      = 997.4968671630002
mass = 997.4968671630002 * (1 - 0.005)^0.5
     = 995
```
Not a big difference here, but those errors would add up!

I fixed it and added tests to verify this property. **Thank you** for finding this!